File: platformer/src/main/java/com/mtgames/platformer/entities/FreeCamera.java
package com.mtgames.platformer.entities;

import com.mtgames.platformer.InputHandler;
import com.mtgames.platformer.gfx.Screen;
import com.mtgames.platformer.scripting.JythonFactory;
import com.mtgames.platformer.scripting.interfaces.EntityInterface;
import com.mtgames.platformer.settings.Properties;
import com.sun.javafx.geom.Vec4f;

import static com.mtgames.platformer.settings.Settings.*;

public class FreeCamera extends AdvancedEntity {

	private final EntityInterface entityInterface;

	public FreeCamera(int x, int y, Properties properties) {
		super(properties, x, y);

		entityInterface = (EntityInterface) JythonFactory.getJythonObject("com.mtgames.platformer.scripting.interfaces.EntityInterface", "entities/FreeCamera.py");
		entityInterface.init(this);
	}

	public void tick() {
		entityInterface.tick(this);
	}

	public void render(Screen screen) {
		entityInterface.render(this, screen);
	}
}

File: platformer/src/main/resources/python/entities/freeCamera.py
from com.mtgames.platformer.scripting.interfaces import EntityInterface
from com.mtgames.platformer.gfx import Screen
from com.mtgames.platformer.entities import LightSource
from com.mtgames.platformer.settings import Settings

from com.sun.javafx.geom import Vec4f

class FreeCamera(EntityInterface):
    def __init__(self):
        self.input = 0
        self.speed = 4

        self.lightSource = 0

    def init(self, entity):
        self.input = entity.getProperties().getInput()

        self.lightSource = LightSource(entity.x, entity.y, entity.getProperties())
        # entity.getProperties().getLevel().addLightSource(self.lightSource)

    def tick(self, entity):
        if self.input.isPressed(Settings.KEY_MOD):
            self.speed = 8
        else:
            self.speed = 4

        if self.input.isPressed(Settings.KEY_LEFT):
            entity.x -= self.speed

        if self.input.isPressed(Settings.KEY_RIGHT):
            entity.x += self.speed

        if self.input.isPressed(Settings.KEY_UP):
            entity.y -= self.speed

        if self.input.isPressed(Settings.KEY_DOWN):
            entity.y += self.speed

        # self.lightSource.move(entity.x, entity.y)

    def render(self, entity):
        Screen.drawRectangle(entity.x-4-Screen.xOffset, entity.y-4-Screen.yOffset, entity.x+4-Screen.xOffset, entity.y+4-Screen.yOffset, Vec4f(0, 0, 0, 0.5))
